### PR TITLE
Create get_initial_params method instead of individual methods for each param

### PR DESCRIPTION
**Pull request by colincadams**
This also allows for custom headers to be set in params.

### CODE REVIEW

File: recidiviz/ingest/us_fl_hendry/us_fl_hendry_scraper.py [hunk 1]
```diff
@@ -95,14 +95,11 @@ def populate_data(self, content, params, ingest_info):
...
-        task_type = params.get('task_type', self.get_initial_task_type())
-        endpoint = params.get('endpoint', self.get_initial_task_type())
-
-        if self.is_initial_task(task_type):
+        if self.is_initial_task(params['task_type']):
             return [self._get_search_page()]
 
         tasks = []
-        tasks.extend(self._get_next_page(content, endpoint))
+        tasks.extend(self._get_next_page(content, params.get('endpoint')))
```
Comment (arian487) [inline recidiviz/ingest/us_fl_hendry/us_fl_hendry_scraper.py]: Can we just do params['endpoint'] here to stay consistent?

Comment (colincadams) [reply recidiviz/ingest/us_fl_hendry/us_fl_hendry_scraper.py]: Yep, this region always sets `endpoint` so that will work!

**Comment by colincadams** [comment]
`create_scraper` doesn't override any of these methods afaict

**Comment by colincadams** [comment]
We could add the new method to it however, if we think that would be useful? I was thinking of creating a separate PR that removes `set_initial_vars` as it isn't clear to me how it is useful. Then I can remove that from the scraper creation docs and instead mention this new method to override initial params.

**Pull request merged**